--- oms-0.1.2/configure.in.orig	Thu Feb 15 02:29:36 2001
+++ oms-0.1.2/configure.in	Wed Jan  2 15:16:42 2002
@@ -54,15 +54,12 @@
 dnl
 dnl ==========
 
-CFLAGS="-pipe -Wall -Werror -D__OMS__ -D_FILE_OFFSET_BITS=64 -D_REENTRANT"
+CFLAGS="$CFLAGS -pipe -Wall -Werror -D__OMS__ -D_FILE_OFFSET_BITS=64 -D_REENTRANT"
 
 AC_ARG_ENABLE(devel,
 	[  --enable-devel	Enable development build [default=no]], enable_devel=yes , enable_devel=no)
 if test "x$enable_devel" = "xyes"; then
-	#CFLAGS="$CFLAGS -g3 -fprofile-arcs -ftest-coverage -pg"
-	CFLAGS="$CFLAGS -g3 -pg -DDEVEL -DDEBUG"
-else
-	CFLAGS="$CFLAGS -O6 -fomit-frame-pointer -funroll-all-loops"
+	CFLAGS="$CFLAGS  -DDEVEL -DDEBUG"
 fi	
 
 
@@ -148,20 +145,17 @@
 	esac	
 	AC_DEFINE(ARCH_X86)
 	arch_type=x86
-	CFLAGS="$CFLAGS -march=$target_cpu"
 	;;
 powerpc)
 	arch_type=ppc
-#	CFLAGS="$CFLAGS -march=ppc"
 	;;	
 ppc)
 	arch_type=ppc
-#	CFLAGS="$CFLAGS -march=ppc"
 	;;	
 alpha*)
 	;;
 sparc)
-	CFLAGS="$CFLAGS -Wcast-align -mcpu=ultrasparc"
+	CFLAGS="$CFLAGS -Wcast-align"
 	;;
 esac
 
